docs/20_smart-contracts/10_getting-started/10_dune-guide/index.md

```diff
@@ -0,0 +1,634 @@
+-- Detecting C compiler ABI info - done
+-- Detecting C compile features
+-- Detecting C compile features - done
+-- Check for working CXX compiler: /usr/bin/c++
+-- Check for working CXX compiler: /usr/bin/c++ -- works
+-- Detecting CXX compiler ABI info
+-- Detecting CXX compiler ABI info - done
+-- Detecting CXX compile features
+-- Detecting CXX compile features - done
+-- Configuring done
+-- Generating done
+-- Build files have been written to: /host/home/lupaxel/work/contracts/hello/build
+Scanning dependencies of target hello_project
+[ 11%] Creating directories for 'hello_project'
+[ 22%] No download step for 'hello_project'
+[ 33%] No patch step for 'hello_project'
+[ 44%] No update step for 'hello_project'
+[ 55%] Performing configure step for 'hello_project'
+-- Setting up CDT Wasm Toolchain 3.1.0 at /usr
+-- Setting up CDT Wasm Toolchain 3.1.0 at /usr
+-- The C compiler identification is Clang 9.0.1
+-- The CXX compiler identification is Clang 9.0.1
+-- Detecting C compiler ABI info
+-- Detecting C compiler ABI info - failed
+-- Detecting C compile features
+-- Detecting C compile features - done
+-- Detecting CXX compiler ABI info
+-- Detecting CXX compiler ABI info - failed
+-- Detecting CXX compile features
+-- Detecting CXX compile features - done
+-- Configuring done
+-- Generating done
+-- Build files have been written to: /host/home/lupaxel/work/contracts/hello/build/hello
+[ 66%] Performing build step for 'hello_project'
+Scanning dependencies of target hello
+[ 50%] Building CXX object CMakeFiles/hello.dir/hello.obj
+Warning, empty ricardian clause file
+[100%] Linking CXX executable hello.wasm
+[100%] Built target hello
+[ 77%] No install step for 'hello_project'
+[ 88%] No test step for 'hello_project'
+[100%] Completed 'hello_project'
+[100%] Built target hello_project
+```
+
+Now you will deploy the contract.
+
+### Step 4 - Deploy Contract
+
+Run the following command to deploy your `hello` contract:
+
+```shell
+dune --deploy build/hello hello
+```
+
+The command will deploy the compiled contract residing at `build/hello` to account `hello`:
+
+```
+#         eosio <= eosio::setcode               {"account":"hello","vmtype":0,"vmversion":0,"code":"0061736d0100000001410d60017e0060017f0060027f7f00...
+#         eosio <= eosio::setabi                {"account":"hello","abi":"0e656f73696f3a3a6162692f312e3200010268690001026e6d046e616d6501000000000000...
+```
+
+You can check that the contract was indeed deployed by running the following command:
+
+```shell
+dune -- cleos get code hello
+```
+
+The command sends the name of code account, `hello`, where the contract was deployed, to the `get_code` endpoint, which replies with the hash of the deployed contract:
+
+```
+code hash: 2737de70263c4ac5ac8e04760e5d7426a8d7c9a1680f2bcc3e44354185badc34
+```
+
+If the code hash is non-zero, as above, then the contract was deployed successfully. You can now test your smart contract.
+
+## Smart Contract Testing
```

Comment:
I don't know why but I was under the impression that DUNE had a test harness for automated testing. Since it doesn't from what I can tell, let's just rename this section to something like: "Interacting with smart contracts" since saying it's really about testing is misleading, and I don't want to get into writing unit tests in other languages that call DUNE commands in this tutorial.

Reply:
Changed